sympy/geometry/ellipse.py

```diff
@@ -327,16 +328,12 @@ def circumference(self):
         >>> p1 = Point(0, 0)
         >>> e1 = Ellipse(p1, 3, 1)
         >>> e1.circumference
-        12*Integral(sqrt((-8*_x**2/9 + 1)/(-_x**2 + 1)), (_x, 0, 1))
+        12*elliptic_e(2*sqrt(2)/3)
 
         """
-        from sympy import Integral
         if self.eccentricity == 1:
-            return 2*pi*self.hradius
-        else:
-            x = Dummy('x', real=True)
-            return 4*self.major*Integral(
-                sqrt((1 - (self.eccentricity*x)**2)/(1 - x**2)), (x, 0, 1))
+            return 2 * pi * self.hradius
```

Comment:
This doesn't look right. Eccentricity == 1 means degenerate ellipse, so not really an ellipse (zero minor axis). The formula with `2*pi` would apply for the circle, `eccentricity == 0` . I recognize you didn't put that formula in... in current master, 
```
>>> e = Ellipse(Point(0, 0), hradius=1, eccentricity=1)
>>> [e.major, e.minor, e.circumference]
[1, 0, 2*pi]
```
which is wrong. 

If we are special-casing eccentricity here to avoid calling elliptic_e, then how about: `2*pi*self.hradius` for 0, and `4*self.major` for 1.

Reply:
(4 because the degenerate ellipse is a line segment of length `2*self.major` traversed twice.)